esrally/driver/runner.py

```diff
@@ -659,13 +659,28 @@ def extract_error_details(self, error_details, data):
             error_details.add((data["status"], None))
 
     def error_description(self, error_details):
+        """
+        Generates error description with an arbitrary limit of 5 errors.
+
+        :param error_details: accumulated error details
+        :return: error description
+        """
         error_descriptions = []
-        for status, reason in error_details:
-            if reason:
-                error_descriptions.append(f"HTTP status: {status}, message: {reason}")
+        is_truncated = False
+        for count, error_detail in enumerate(sorted(error_details)):
+            status, reason = error_detail
+            if count < 5:
+                if reason:
+                    error_descriptions.append(f"HTTP status: {status}, message: {reason}")
+                else:
+                    error_descriptions.append(f"HTTP status: {status}")
             else:
-                error_descriptions.append(f"HTTP status: {status}")
-        return " | ".join(sorted(error_descriptions))
+                is_truncated = True
```

Comment:
What do you think about summarising the remaining errors? E.g Truncated 20 X 409, 30 X 503 etc? That way we can easily see if the errors we got back were at least all of the same status code, or whether there was other statuses hidden further along?

Reply:
Great idea, let me add this :)

Reply:
Added in https://github.com/elastic/rally/pull/1835/commits/6ce175adbd18fe48dc461ed8fc7356156d5dac3f.